### DISCUSSION TITLE
SnapCast Support

### DISCUSSION THREAD

**Discussion by sturd** [jcorporation/myMPD#910] [Ideas]
I'm really enjoying using myMPD, so far.  Thank you for the work you've done on it!

The one main feature that I'm missing in myMPD - and any MPD client, for that matter - is support for controlling volumes of clients in SnapCast.  Having to switch to a different (web) app to manage that is quite clunky.

I've forked myMPD, and started making some exploratory commits in a branch off `devel `.
https://github.com/sturd/myMPD/tree/sturd_snapcast_enhance

Would there be interest in merging it in to the one-true-repo, if I make good progress with it?

**Comment by jcorporation** [comment]
What is the exact usecase? MPD already includes a snapcast output.

**Reply by sturd** [reply]
You're free to output the audio to SnapCast directly from MPD, via pipe, or via stdout.  But not many clients offer direct control of SnapCast itself, meaning you have to resort to something like [the web app](https://github.com/badaix/snapcast/#webapp).

I'm proposing/exploring implementation of controls to directly control the volumes/configurations of SnapCast itself, using its own JSON-RPC.  **_Of course, it would be off and hidden, by default!_**

**Reply by sturd** [reply]
Oh, I realise you mean the output which makes MPD the de-facto SnapCast server.  This option would work in the case where MPD is the only input to SnapCast, but that's unfortunately not how it is configured for me.

**Reply by Ohp3ueyi** [reply]
I've got my "master" raspberry attached to screen with myMPD on top of MPD that sends audio via snapcast to may "bedroom" raspberry and "living room" raspberry. 
Now there 2 issues with volume control over MPD - one not being able to control volume on my slave raspberries separately and second, that over MPD i can control only software volume (as both of them are equipped with DACs and snapcast allows hardware volume control (like, yeah, i'm such an audiophile that i could actually hear the difference, but whatever... at least it gives me a purist sense of accomplishment))
I know i'm making claims but my family can make a lot of mess trying to control volumes and i'd be so happy if there were less places to mess things up...
Mopidy + Iris +snapcast gives such possibility but after years of trying different configurations of DIY home audio solutions, MPD + myMPD + snapcast works best for me.

**Comment by jcorporation** [comment]
Sorry, but this feature is out of scope of myMPD.

**Comment by jcorporation** [comment]
If there is a real demand we can try to implement this as an plugin. But at the moment there is no plugin system in myMPD.
